src/main/java/com/employeesService/EmployeesService/service/DepartmentServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.employeesService.EmployeesService.service;
+
+import com.employeesService.EmployeesService.model.Department;
+import com.employeesService.EmployeesService.repository.DepartmentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class DepartmentServiceImpl implements DepartmentService{
+    private DepartmentRepository departmentRepository;
+
+    @Autowired
+    public DepartmentServiceImpl(DepartmentRepository departmentRepository){
+        this.departmentRepository=departmentRepository;
+    }
+
+    @Override
+    public Department save(Department department) {
+        return departmentRepository.save(department);
+    }
+
+    @Override
+    public List<Department> getAll() {
+        return departmentRepository.findAll();
+    }
+
+    @Override
+    public Department getById(String deptNo) {
+        Department department= departmentRepository.findById(deptNo).orElse(null);
```

Comment:
In case the `Department` is not found for the given id, throw a custom exception `DepartmentNotFoundException` which should be mapped to `404` HTTP Status code with the message: `Department not found with id: {theIdHere}`